Define daddr_t if __DADDR_T_TYPE is not defined

glibc defined daddr_t but musl does not, ideally it should not be used
and simple int type is enough. However, its better to leave glibc behavior
as it is and only define it to int if daddr_t is not provided by libc

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
--- a/stress-ng.h
+++ b/stress-ng.h
@@ -3750,6 +3750,10 @@ struct shim_statx {
 	uint64_t __spare2[14];
 };
 
+#ifndef __DADDR_T_TYPE
+typedef int daddr_t;
+#endif
+
 /* old ustat struct */
 struct shim_ustat {
 	daddr_t	f_tfree;
